tic-tac-toe.js

```diff
@@ -1,188 +1,172 @@
-            (this.field[0][2] == "O" & this.field[1][2] == "O" & this.field[2][2] == "O") ||
-            (this.field[0][0] == "O" & this.field[1][1] == "O" & this.field[2][2] == "O") ||
-            (this.field[2][0] == "O" & this.field[1][1] == "O" & this.field[0][2] == "O")
-            ) { this.currentState = states[2];
-                console.log("Player 2 has won. Reiniating the Game.");
-                this.resetGame();
-            }
-        if (!this.field[0].includes(".") & !this.field[1].includes(".") & !this.field[2].includes(".")  )
-            {            
-            this.currentState = states[3];
-            console.log("Draw. Reiniating the Game.");
-            this.resetGame();
-        }    
-        else {
-            this.currentState = states[0]
-        }
-
-    };
-
-    move(a,b) {
-        if (this.field[a][b] == "X" || this.field[a][b] == "O") {
-            console.log("This square is already ocupied");
-            return "This square is already ocupied"
-            }  
-        if (this.currentPlayer == players[0]) {
-            this.field[a][b] = "X";
-            this.currentPlayer = players[1];
-            } 
-        else {
-            this.field[a][b] = "O";
-            this.currentPlayer = players[0];
-            }
-    };
-
-    selectUpLeft() {
-       this.move(0,0);
-       this.gameState()
-    };
-    selectUpCenter() {
-        this.move(0,1);
-        this.gameState()
-    };
-    selectUpRight() {
-        this.move(0,2);
-        this.gameState()
-    };
-    selectCenterLeft() {
-        this.move(1,0);
-        this.gameState()
-     };
-    selectCenterCenter() {
-        this.move(1,1);
-        this.gameState()
-     };
-    selectCenterRight() {
-        this.move(1,2);
-        this.gameState()
-     };
-    selectDownLeft() {
-        this.move(2,0),
-        this.gameState()
-     };
-    selectDownCenter() {
-        this.move(2,1);
-        this.gameState()
-     };
-    selectDownRight() {
-        this.move(2,2);
-        this.gameState()
-     };
-  
-    /*
-     moves(a) {
-         this.movesArray = [this.selectUpLeft(),
-            this.selectUpCenter(),
-            this.selectUpRight(),
-            this.selectCenterLeft(),
-            this.selectCenterCenter(),
-            this.selectCenterRight(),
-            this.selectDownLeft(),
-            this.selectDownCenter(),
-            this.selectDownRight()];
-        this.movesArray[a];
+  _gameState() {
+    if (this._playerHasWon(Marks.P1)) {
+      this.currentState = states.P1;
+      console.log("Player 1 has won.");
+      return;
     }
-    THIS DIDN'T WORK...
 
-    */
-};
+    if (this._playerHasWon(Marks.P2)) {
+      this.currentState = states.P2;
+      console.log("Player 2 has won.");
+      return;
+    }
+
+    if (
+      !this.field[0].includes(Marks.Empty) &
+      !this.field[1].includes(Marks.Empty) &
+      !this.field[2].includes(Marks.Empty)
```

Comment:
Just nice to have, this could also be a function so it is even more easy to read:

```
if (this._boardIsFull()){
...
}

_boardIsFull(){
 return !this.field[0].includes(Marks.Empty) &
      !this.field[1].includes(Marks.Empty) &
      !this.field[2].includes(Marks.Empty)
}

```

Reply:
Done